src/Conversion/KrnlToLLVM/KrnlCall.cpp

```diff
@@ -0,0 +1,214 @@
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===------ KrnlCall.cpp - Lower KrnlCallOp -------------------===//
+//
+// Copyright 2019-2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file lowers the KrnlCallOp operator.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Conversion/LLVMCommon/Pattern.h"
+#include "mlir/Conversion/LLVMCommon/TypeConverter.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+
+#include "src/Conversion/KrnlToLLVM/KrnlToLLVMHelper.hpp"
+#include "src/Dialect/Krnl/DialectBuilder.hpp"
+#include "src/Dialect/Krnl/KrnlHelper.hpp"
+#include "src/Dialect/Krnl/KrnlOps.hpp"
+#include "llvm/Support/Debug.h"
+
+#define DEBUG_TYPE "krnl_to_llvm"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+namespace krnl {
+
+class KrnlCallOpLowering : public ConversionPattern {
+public:
+  explicit KrnlCallOpLowering(
+      TypeConverter &typeConverter, MLIRContext *context)
+      : ConversionPattern(
+            typeConverter, KrnlCallOp::getOperationName(), 1, context) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    KrnlCallOpAdaptor krnlCallAdaptor(operands);
+    auto loc = op->getLoc();
+    KrnlCallOp krnlCallOp = llvm::dyn_cast<KrnlCallOp>(op);
```

Comment:
I think here we can use `llvm::cast` given we are not checking whether `krnlCallOp` is null